components/bio-formats-plugins/src/loci/plugins/util/ROIHandler.java

```diff
@@ -301,17 +306,13 @@ else if (shapeObject instanceof ome.xml.model.Rectangle) {
 
                     if (roi != null) {
                         roi.setName(shapeObject.getID());
-                        
-                        if (shapeObject.getTheC() != null &&
-                          shapeObject.getTheZ() != null &&
-                          shapeObject.getTheT() != null)
-                        {
-                          int c = shapeObject.getTheC().getValue();
-                          int z = shapeObject.getTheZ().getValue();
-                          int t = shapeObject.getTheT().getValue();
-                          roi.setPosition(c, z, t);
-                        }
 
+                        if (Prefs.showAllSliceOnly){
+                            if(shapeObject.getTheC() != null);c = shapeObject.getTheC().getValue();
+                            if(shapeObject.getTheZ() != null);z = shapeObject.getTheZ().getValue();
+                            if(shapeObject.getTheT() != null);t = shapeObject.getTheT().getValue();
```

Comment:
The semi-colons after the ```if(...)``` statements are probably not what you want here (see https://ci.openmicroscopy.org/view/Bio-Formats/job/BIOFORMATS-5.1-merge-build/782/findbugsResult/new/HIGH/).  Something like:

```
if (shapeObject.getTheC() != null) {
  c = shapeObject.getTheC().getValue();
}
```

addresses the formatting comment from @jburel and preserves the original logic.